tests/test_rank.py

```diff
@@ -160,3 +160,105 @@ def test_order_label_issues_using_self_confidence_ranking():
     score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
     label_issues_indices2 = [z[1] for z in score_idx]
     assert all(label_issues_indices == label_issues_indices2)
+
+
+def test_order_label_issues_using_confidence_weighted_entropy_ranking():
+    label_issues_indices = rank.order_label_issues(
+        label_issues_mask=data["label_errors_mask"],
+        labels=data["labels"],
+        pred_probs=data["pred_probs"],
+        rank_by="confidence_weighted_entropy",
+    )
+    scores = rank.get_confidence_weighted_entropy_for_each_label(data["labels"], data["pred_probs"])
+    indices = np.arange(len(scores))[data["label_errors_mask"]]
+    scores = scores[data["label_errors_mask"]]
+    score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
+    label_issues_indices2 = [z[1] for z in score_idx]
+    assert all(label_issues_indices == label_issues_indices2)
+
+
+def test_order_label_issues_using_scoring_func_ranking():
+
+    # test all scoring methods with the scoring function
+    scoring_methods = ["self_confidence", "normalized_margin", "confidence_weighted_entropy"]
+
+    for method in scoring_methods:
+        label_issues_indices = rank.order_label_issues(
+            label_issues_mask=data["label_errors_mask"],
+            labels=data["labels"],
+            pred_probs=data["pred_probs"],
+            rank_by=method,
+        )
+        scores = rank.score_label_quality(data["labels"], data["pred_probs"], method=method)
+        indices = np.arange(len(scores))[data["label_errors_mask"]]
+        scores = scores[data["label_errors_mask"]]
+        score_idx = sorted(list(zip(scores, indices)), key=lambda y: y[0])  # sort indices by score
+        label_issues_indices2 = [z[1] for z in score_idx]
+        assert all(label_issues_indices == label_issues_indices2)
+
+
+def test_order_label_issues_using_scoring_func_adj_pred_probs_ranking():
```

Comment:
To me, this seems like similar test function as: `test_order_label_issues_using_confidence_weighted_entropy_ranking()` except just need to pass in special `rank_by_kwargs`.  Could we reduce copy/pasted code, by defining one helper function which can optionally take in `rank_by_kwargs` and pass it where it's needed, and then just have both this test and `test_order_label_issues_using_confidence_weighted_entropy_ranking()` call the helper function with different args?  Or is it better to keep these two tests as entirely different functions without any shared helper method?

Reply:
@JohnsonKuan one way to do this is to use @paramterize and pass parameters to create separate tests for every method without re-writing code. Example of how to do that here: https://github.com/cleanlab/cleanlab/blob/master/tests/test_filter_count.py#L332

Reply:
Agree, good catch. I removed this test function. This is now taken care of in the other test function, `test_order_label_issues_using_scoring_func_ranking`, where I added @parameterize decorator to pass the rank_by arg.

Thanks @cgnorthcutt for the tip on @parameterize

315e956b18f489a90b90a2ca62016b34176997fc

Reply:
Given the switch to `parameterize` decorator based testing, do we still need the singleton tests:
`test_get_normalized_margin_for_each_label()`
`test_get_self_confidence_for_each_label()` ?

I suspect no, so let's delete these.  But if yes, then why is there no such test for the new scoring funcs?

Reply:
@cgnorthcutt Will let you review the `parameterize` decorator based tests.